src/Playwright/Helpers/TaskHelper.cs

```diff
@@ -133,15 +133,10 @@ public static Task<T> WithTimeout<T>(this Task<T> task, Action timeoutAction, in
     /// <typeparam name="T">Return type.</typeparam>
     public static async Task<T> WithTimeout<T>(this Task<T> task, Action timeoutAction, TimeSpan timeout)
     {
-        if (task == null)
-        {
-            return default;
-        }
-
         if (await TimeoutTask(task, timeout).ConfigureAwait(false) && timeoutAction != null)
         {
             timeoutAction();
-            return default;
+            return default!;
```

Comment:
Maybe leave a comment that this line is unreachable? Or better yet `throw timeoutAction();` or something like that, so that you don't have to add unreachable return statement.